internal/bminventory/inventory.go

```diff
@@ -1647,12 +1647,14 @@ func (b *bareMetalInventory) updateNonDhcpNetworkParams(updates map[string]inter
 	}
 
 	var err error
-	*machineCidr, err = network.CalculateMachineNetworkCIDR(apiVip, ingressVip, cluster.Hosts)
+	*machineCidr, err = network.CalculateMachineNetworkCIDR(apiVip, ingressVip, cluster.Hosts, false)
 	if err != nil {
 		log.WithError(err).Errorf("failed to calculate machine network cidr for cluster: %s", params.ClusterID)
 		return common.NewApiError(http.StatusBadRequest, err)
 	}
-	setMachineNetworkCIDRForUpdate(updates, *machineCidr)
+	if *machineCidr != "" {
```

Comment:
Why is this correct?

Reply:
not sure what do you mean, cidr can return as an empty string in case we don't have enough data, why do we need to enter `setMachineNetworkCIDRForUpdate` in this case?

Reply:
Maybe we want to clear ?